shared-module/storage/__init__.c

```diff
@@ -177,15 +177,16 @@ void common_hal_storage_mount(mp_obj_t vfs_obj, const char *mount_path, bool rea
     args[0] = readonly ? mp_const_true : mp_const_false;
     args[1] = mp_const_false; // Don't make the file system automatically when mounting.
 
-    // Check that there's no file or directory with the same name as the mount point.
+    // Check that there's file or directory with the same name as the mount point.
     // But it's ok to mount '/' in any case.
     if (strcmp(vfs->str, "/") != 0) {
         nlr_buf_t nlr;
         if (nlr_push(&nlr) == 0) {
             common_hal_os_stat(mount_path);
             nlr_pop();
-            // Something with the same name exists.
-            mp_raise_OSError(MP_EEXIST);
+        } else {
+            // Something with the same name doesn't exists.
+            mp_raise_RuntimeError(MP_ERROR_TEXT("Mount point missing. Create first (maybe via USB)"));
```

Comment:
What's the consequence of making this a warning in 9 and an error in 10? Maybe something like this but less horribly verbose.
```
warnings_warn(&mp_type_FutureWarning, MP_ERROR_TEXT(
    "Mount point missing. "
    "This will be an error in CircuitPython 10. "
    "This prevents file from being available via web & bluetooth workflow. "
    "Create an empty directory (maybe via USB) first");
```

Also is it important that the placeholder be a directory, which is a requirement on Unix-style systems I have used?

Reply:
If it is a warning, then the parent directory won't list the mount point. So navigating into the mount wouldn't work.